File: backend/app/schemas/evento.py
from datetime import datetime
from typing import Literal
from pydantic import BaseModel, ConfigDict


TipoEvento = Literal["academico", "deportivo", "cultural", "administrativo", "otro"]


class EventoBase(BaseModel):
    """Atributos base de un evento programado."""
    espacio_id: int | None = None
    titulo: str
    descripcion: str | None = None
    fecha_inicio: datetime
    fecha_fin: datetime | None = None
    tipo: TipoEvento = "otro"
    foto_url: str | None = None
    url_registro: str | None = None
    activo: bool = True


class EventoCreate(EventoBase):
    """Esquema para creación de eventos."""
    pass


class EventoUpdate(BaseModel):
    """Campos actualizables del evento."""
    espacio_id: int | None = None
    titulo: str | None = None
    descripcion: str | None = None
    fecha_inicio: datetime | None = None
    fecha_fin: datetime | None = None
    tipo: TipoEvento | None = None
    foto_url: str | None = None
    url_registro: str | None = None
    activo: bool | None = None


class EventoOut(EventoBase):
    """Esquema de salida para eventos."""
    model_config = ConfigDict(from_attributes=True)

    id: int
    creado_en: datetime

File: frontend/src/types/evento.ts
export type TipoEvento = "academico" | "deportivo" | "cultural" | "administrativo" | "otro";

export interface Evento {
    id: number;
    espacio_id: number | null;
    titulo: string;
    descripcion: string | null;
    fecha_inicio: string;
    fecha_fin: string | null;
    tipo: TipoEvento;
    foto_url: string | null;
    url_registro: string | null;
    activo: boolean;
    creado_en: string;
}

export type EventoCreate = Omit<Evento, "id" | "creado_en">;
export type EventoUpdate = Partial<EventoCreate>;
